sweep/sweeper.go

```diff
@@ -454,6 +644,60 @@ func (s *UtxoSweeper) collector(blockEpochs <-chan *chainntnfs.BlockEpoch,
 	}
 }
 
+// bucketForFeeReate determines the proper bucket for a fee rate. This is done
+// in order to batch inputs with similar fee rates together.
+func (s *UtxoSweeper) bucketForFeeRate(
+	feeRate lnwallet.SatPerKWeight) lnwallet.SatPerKWeight {
+
+	step := s.cfg.FeeRateSpacing
+	minBucket := s.cfg.MinFeeRate + step
+	for bucket := minBucket; bucket <= s.cfg.MaxFeeRate; bucket += step {
+		if feeRate <= bucket {
+			return bucket
+		}
+	}
+
+	// This should never happen as we assume that the fee rate has already
+	// been validated so that it fits within our constraints.
+	return minBucket
+}
+
+// clusterBySweepFeeRate takes the set of pending inputs within the UtxoSweeper
+// and clusters those together with similar fee rates. Each cluster contains a
+// sweep fee rate, which is determined by calculating the average fee rate of
+// all inputs within that cluster.
+func (s *UtxoSweeper) clusterBySweepFeeRate() []inputCluster {
+	// First, we'll group together all inputs with similar fee rates. This
+	// is done by determining the fee rate bucket they should belong in.
+	bucketInputs := make(map[lnwallet.SatPerKWeight]pendingInputs)
+	for _, input := range s.pendingInputs {
+		bucket := s.bucketForFeeRate(input.feeRate)
```

Comment:
With the fee rate callback suggestion above, this would become `bucket := s.bucketForFeeRate(input.getFeeRate())`